Add helper manual (-h --help) option

imperivox opened the pull request:
This should work as a helper manual.

imperivox committed:
config.rs

-h or --help feature added into config

imperivox committed:
Add helper manual (-h) option

- Add helper manual text

arthurbacci commented on src/config.rs:
```diff
@@ -3,9 +3,18 @@ use std::env;
 pub struct Config {
     pub file_path: String,
     pub little_endianess: bool,
+    pub show_help: bool,
```
Not the most idiomatic way to do this.

arthurbacci replied:
Fixed

arthurbacci commented on src/config.rs:
```diff
@@ -3,9 +3,18 @@ use std::env;
 }
 impl Config {
     pub fn new(args: &[String]) -> Result<Config, &'static str> {
+        if args.contains(&"-h".to_string()) || args.contains(&"--help".to_string()) {
```
Fine, but I think this may be better treated as an error.

arthurbacci replied:
Done

arthurbacci commented on src/main.rs:
```diff
@@ -2,12 +2,56 @@ use byteblitz::config::Config;
 use std::process;
 
+fn print_help_manual() {
+    let help_text = format!(
```
Good, thank you a lot for writing all this

arthurbacci commented on src/main.rs:
```diff
@@ -2,12 +2,56 @@ use byteblitz::config::Config;
         println!("Problem parsing arguments: {err}");
         process::exit(1);
     });
+    if config.show_help {
```
Good

arthurbacci reviewed (changes requested):
Return an error instead of making changes to the Config structure.

arthurbacci committed:
refactor: return error instead of another variant of Config

arthurbacci commented:
> Return an error instead of making changes to the Config structure.

Done

noahra commented:
LGTM 🍰

imperivox committed:
main.rs

added to.string() instead of format! because of the error: useless use of `format!` error on ubuntu
